src/shared/scraper.js

```diff
@@ -0,0 +1,74 @@
+
+
+/** @typedef Product
+ *  @property dom  DOM element that contains the relevant data
+ *  @property {number} rating  Average rating. If there are not reviews, this will be null.
+ *  @property {number} reviewCount  Number of reviews
+ *  @property {array} distributions  Fractions of each star rating over all ratings. Indices 0-5 => 1-5 stars
+ */
+
+
+export default {
+
+    /**
+     * Find all products on the page and extract some basic stats
+     * @returns {Promise<Product>}
+     */
+    async findAllProducts(){
+        return new Promise((resolve, reject) => {
+            if (document.readyState == "loading"){
+                document.addEventListener('DOMContentLoaded', run)
+            } else {
+                run()
+            }
+            function run(){
+                //TODO sometimes products are featured together, and this selects the entire block instead of each product
+                const images = document.querySelectorAll('[data-component-type=s-product-image]')
```

Comment:
Nice, I didn't know you could select based on attributes

Reply:
:grin: